protocol/communities_events_eventual_consistency_test.go

```diff
@@ -0,0 +1,178 @@
+	community := setUpOnRequestCommunityAndRoles(s, protobuf.CommunityMember_ROLE_ADMIN, []*Messenger{})
+
+	// set up additional user that will send request to join
+	user := s.newMessenger("", []string{})
+	s.SetupAdditionalMessengers([]*Messenger{user})
+
+	advertiseCommunityToUserOldWay(&s.Suite, community, s.owner, user)
+
+	// user sends request to join
+	requestToJoin := &requests.RequestToJoinCommunity{CommunityID: community.ID(), ENSName: "testName"}
+	response, err := user.RequestToJoinCommunity(requestToJoin)
+	s.Require().NoError(err)
+	s.Require().NotNil(response)
+	s.Require().Len(response.RequestsToJoinCommunity, 1)
+
+	sentRequest := response.RequestsToJoinCommunity[0]
+
+	checkRequestToJoin := func(r *MessengerResponse) bool {
+		for _, request := range r.RequestsToJoinCommunity {
+			if request.ENSName == requestToJoin.ENSName {
+				return true
+			}
+		}
+		return false
+	}
+
+	// admin receives request to join
+	response, err = WaitOnMessengerResponse(
+		s.admin,
+		checkRequestToJoin,
+		"event sender did not receive community request to join",
+	)
+	s.Require().NoError(err)
+	s.Require().Len(response.RequestsToJoinCommunity, 1)
+
+	// accept request to join
+	acceptRequestToJoin := &requests.AcceptRequestToJoinCommunity{ID: sentRequest.ID}
+	_, err = s.admin.AcceptRequestToJoinCommunity(acceptRequestToJoin)
+	s.Require().NoError(err)
+
+	// then reject request to join
+	rejectRequestToJoin := &requests.DeclineRequestToJoinCommunity{ID: sentRequest.ID}
+	_, err = s.admin.DeclineRequestToJoinCommunity(rejectRequestToJoin)
+	s.Require().NoError(err)
+
+	// ensure both messages are pushed to waku
+	/*
+		FIXME: we should do it smarter, as follows:
+		```
+		hashes1, err := admin.SendEvent()
+		hashes2, err := admin.SendEvent()
+		WaitForHashes([][]byte{hashes1, hashes2}, admin.waku)
+		s.messagesOrderController.setCustomOrder([][]{hashes1, hashes2})
+		```
+	*/
+	time.Sleep(1 * time.Second)
+
+	// ensure events are received in order
+	s.messagesOrderController.order = messagesOrderInPostOrder
+
+	waitForAcceptedRequestToJoin := waitOnCommunitiesEvent(s.owner, func(sub *communities.Subscription) bool {
+		return len(sub.AcceptedRequestsToJoin) == 1
+	})
+
+	waitOnAdminEventsRejection := waitOnCommunitiesEvent(s.owner, func(s *communities.Subscription) bool {
+		return s.CommunityEventsMessageInvalidClock != nil
+	})
+
+	_, err = s.owner.RetrieveAll()
+	s.Require().NoError(err)
+
+	// first owner handles AcceptRequestToJoinCommunity event
+	err = <-waitForAcceptedRequestToJoin
+	s.Require().NoError(err)
+
+	// then owner rejects DeclineRequestToJoinCommunity event due to invalid clock
+	err = <-waitOnAdminEventsRejection
+	s.Require().NoError(err)
+
+	// admin receives rejected DeclineRequestToJoinCommunity event and re-applies it,
+	// there is no signal whatsoever, we just wait for admin to process all incoming messages
+	_, _ = WaitOnMessengerResponse(s.admin, func(response *MessengerResponse) bool {
+		return false
+	}, "")
```

Comment:
Should we move this under `communityRequestsEventualConsistencyFixed` condition as well?